### PR TITLE
feat: Check catalog availability before creating

### PR DESCRIPTION
**Pull request by Xuanwo**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR add checks while creating catalog to make sure the catalog is valid.

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13415)

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_catalog_create.rs [hunk 1]
```diff
@@ -59,6 +63,25 @@ impl Interpreter for CreateCatalogInterpreter {
...
+                name_ident: CatalogNameIdent {
+                    tenant: self.plan.tenant.clone(),
+                    catalog_name: self.plan.catalog.clone(),
+                },
+                meta: CatalogMeta {
+                    catalog_option: self.plan.meta.catalog_option.clone(),
+                    created_on: chrono::Utc::now(),
+                },
+            })
+            .map_err(|err| err.add_message("The catalog you're trying to create is invalid."))?;
```
Comment (soyeric128) [inline src/query/service/src/interpreters/interpreter_catalog_create.rs:80]: The catalog you're trying to create is invalid.-> Error creating catalog.

File: src/query/service/src/interpreters/interpreter_catalog_create.rs [hunk 2]
```diff
@@ -59,6 +63,25 @@ impl Interpreter for CreateCatalogInterpreter {
...
+
+        // list databases to check if the catalog is valid.
+        let _ = ctl.list_databases(&self.plan.tenant).await.map_err(|err| err.add_message("The catalog creation failed the validation check. Please verify if its configuration is valid."))?;
```
Comment (soyeric128) [inline src/query/service/src/interpreters/interpreter_catalog_create.rs:83]: Catalog creation failed. Check your parameter values.

Comment (Xuanwo) [reply src/query/service/src/interpreters/interpreter_catalog_create.rs]: addressed.

File: src/query/service/src/interpreters/interpreter_catalog_create.rs (hunk 1 above)
Comment (Xuanwo) [reply src/query/service/src/interpreters/interpreter_catalog_create.rs]: addressed.

**Comment by BohuTANG** [comment]
```
0: statement failed: mysql client error: Server error: `ERROR HY000 (1105): StorageUnavailable. Code: 3901, Text = Catalog creation failed. Check your parameter values.
current configured storage is not available: config: Fs(StorageFsConfig { root: "/root/" }), cause: PermissionDenied (persistent) at list, context: { service: fs, path: / } => permission denied, source: Permission denied (os error 13).'
[SQL] CREATE CATALOG ctl TYPE=ICEBERG CONNECTION=( URL='fs://tmp' );
Error: SelfError("sqllogictest failed")
at tests/sqllogictests/suites/base/05_ddl/05_0
```

**Review by flaneur2020 — APPROVED** [review]
LGTM

**Pull request merged**